dbms/src/Flash/Coprocessor/DAGQueryBlockInterpreter.cpp

```diff
@@ -451,6 +451,17 @@ void DAGQueryBlockInterpreter::recordProfileStreams(DAGPipeline & pipeline, cons
     pipeline.transform([&profile_streams](auto & stream) { profile_streams.push_back(stream); });
 }
 
+void DAGQueryBlockInterpreter::recordProfileStream(const BlockInputStreamPtr & stream, size_t concurrency, const String & key)
+{
+    assert(stream);
+    auto & profile_streams = dagContext().getProfileStreamsMap()[key];
+    profile_streams.push_back(stream);
+    const auto & header = stream->getHeader();
+    /// We consider the number of streams as the concurrency of the operator and pass it to tidb.
+    for (size_t i = 1; i < concurrency; ++i)
+        profile_streams.emplace_back(std::make_shared<NullBlockInputStream>(header));
```

Comment:
Do we fill the concurrency here?
https://github.com/pingcap/tiflash/blob/f97567ab67355768059fa93bc445e8ac5dbe5a39/dbms/src/Flash/Coprocessor/DAGResponseWriter.cpp#L94-L103

I don't get it that why using `NullBlockInputStream` here won't affect the number of actual rows, actual process time, etc.

Reply:
because `NullBlockInputStream ` is not `IProfilingBlockInputStream`..